qutebrowser/components/adblock.py

```diff
@@ -206,53 +238,26 @@ def _import_local(self, filename: str) -> None:
         self._in_progress.append(download)
         self._on_download_finished(download)
 
-    def _parse_line(self, raw_line: bytes) -> bool:
+    def _parse_line(self, raw_line: bytes) -> list:
         """Parse a line from a host file.
 
         Args:
             raw_line: The bytes object to parse.
 
         Returns:
-            True if parsing succeeded, False otherwise.
+            A list containg parsed hosts.
         """
         if raw_line.startswith(b'#'):
```

Comment:
Maybe this logic (checking for comments early and decoding lines) should also be moved into `_read_hosts_line`? At that point, you can probably get rid of this function entirely, as it's just `self._blocked_hosts.update(self._read_hosts_line(line))`.

Reply:
_read_hosts_line is used when parsing a line from either a local hosts file (string) or from a downloaded hosts list (bytes object). Decoding is only needed when downloading a list (in _parse_line). _read_hosts_line could be made to do the decoding, but then either a) it would need to take both strings as well as bytes objects or b) _read_hosts_file would need to be changed so that it gives a bytes object to _read_hosts_line as well. To me, both solutions seem like they would make parts of the code messier for the sake of making other parts tidier. If you think otherwise I'm happy to do whichever you think is best.

Even if no change is needed, the content of _parse_line is already fairly small so it could be done away with and the decoding could just be done in _merge_file where _parse_line is being called.

Reply:
> b) _read_hosts_file would need to be changed so that it gives a bytes object to _read_hosts_line as well.

All that should be needed there is changing `with open(filename, 'r', encoding='utf-8')` to `with open(filename, 'rb')`, no?

Reply:
My bad, I'm not as familiar with dealing with bytes objects :P 
I will do that.